synapse/tools/autodoc.py

```diff
@@ -291,49 +303,156 @@
+                            if i == 0:
+                                rst.addLines(f'        - | {k}: ``{v}``')
+                            else:
+                                rst.addLines(f'          | {k}: ``{v}``')
+                else:
+                    rst.addLines(f'        - ')
+
+        if formedges:
+
+            source_edges = formedges.pop('source', None)
+            dst_edges = formedges.pop('target', None)
+            generic_edges = formedges.pop('generic', None)
+
+            if source_edges:
+                if generic_edges:
+                    source_edges.extend(generic_edges)
+
+                rst.addLines(f'  Source Edges:',)
+                rst.addLines('   .. list-table::',
+                             '      :header-rows: 1',
+                             '      :widths: 10 15 10 65',
+                             '      :class: tight-table',
+                             '',
+                             '      * - source',
+                             '        - verb',
+                             '        - target',
+                             '        - doc',
                              )
-                for k, v in popts.items():
-                    k = poptsToWords.get(k, k.replace(':', raw_back_slash_colon))
-                    rst.addLines('  ' + f'* {k}: ``{v}``')
 
-            hptlink = f'dm-type-{ptname.replace(":", "-")}'
-            tdoc = f'The property type is :ref:`{hptlink}`.'
+                _edges = []
+                for (edef, enfo) in source_edges:
+                    src, enam, dst = edef
+                    doc = enfo.pop('doc', None)
+
+                    if src is None:
+                        src = '*'
+                    if dst is None:
+                        dst = '*'
+
+                    if enfo:
+                        logger.warning(f'{name} => Light edge {enam} has unhandled info: {enfo}')
+                    _edges.append((src, enam, dst, doc))
+                _edges.sort(key=lambda x: x[:2])
+
+                for src, enam, dst, doc in _edges:
+                    rst.addLines(f'      * - ``{src}``',
+                                 f'        - ``-({enam})>``',
+                                 f'        - ``{dst}``',
+                                 f'        - {doc}',
+                                 )
+
+            if dst_edges:
+                if generic_edges:
+                    dst_edges.extend(generic_edges)
+
+                # dst_edges.sort(key=lambda x: x[0])
+
+                rst.addLines(f'  Target Edges:', )
+                rst.addLines('   .. list-table::',
+                             '      :header-rows: 1',
+                             '      :widths: 10 15 10 65',
+                             '      :class: tight-table',
+                             '',
+                             '      * - source',
+                             '        - verb',
```

Comment:
verb width seems a little wide; was this for future-proofing or consistency with another table?

Reply:
I originally had it at 10%; but moved it to 15% since we added some built in verbs which are longer and caused line wrapping.

Reply:
Changing the widths to `auto` works fine too.